FHEM/00_SIGNALduino.pm

```diff
@@ -2071,13 +2073,22 @@ SIGNALduino_Parse_MS($$$$%)
 			$valid = $valid && ($pstr=SIGNALduino_PatternExists($hash,\@{$ProtocolListSIGNALduino{$id}{one}},\%patternList,\$rawData)) >=0;
 			Debug "Found matched one with indexes: ($pstr)" if ($debug && $valid);
 			$patternLookupHash{$pstr}="1" if ($valid); ## Append Sync to our lookuptable
+			$endPatternLookupHash{substr($pstr,0,length($pstr)-1)}="1" if ($valid); ## Append Sync to our lookuptable
```

Comment:
Der Kommentar in der Zeile ist bestimmt falsch.
Was ich nicht ganz verstehe, Du machst einen substr von 0 bis zu vorletzten Stelle. Ich nehme an, das ist so, weil der letzte Puls fehlen könnte

Reply:
Genau, wobei der letzte nicht im eigentlichen Sinne Pulse ist, sondern Lücke. Und die ist im letzten Bit identisch mit der Pause bis zur nächsten Sendung, also die freie Zeit im Kanal. Also z.B. 
  100ms Puls + 200ms Pause für one
  200ms Puls + 100ms Pause für Zero
werden erkannt. Als letztes dann
  100ms Puls + lange Pause, die auch nicht mehr im Datensatz steht.
Dann feht bei den Protokollen ohne Stopbit der letzte Eintrag und man kann auch nach der Dekodierung nicht mehr wissen, ob dort als letztes Bit 0 oder 1 stehen soll.

Reply:
Jo danke für die Erläuterung. Hatte ich mir auch so gedacht :)